test/apps/data-filter/src/data-filter-extension.js

```diff
@@ -0,0 +1,119 @@
+import {LayerExtension} from '@deck.gl/core';
+
+const DATA_TYPE_FROM_SIZE = {
```

Comment:
what's the reason of having 4 different data types here?

Reply:
This is so that the app can specify the number of filters to use. Say your earthquake data contains magnitude and time. You will create a data filter of size 2, which allows you to specify independent ranges for each value.

Reply:
We could potentially expand this to support larger sizes but 4 is probably good enough to cover most use cases?